Assets/MixedRealityToolkit-SDK/Features/Audio/Influencers/AudioInfluencerController.cs

```diff
@@ -0,0 +1,313 @@
+            highPassFilter = gameObject.GetComponent<AudioHighPassFilter>();
+            nativeHighPassCutoffFrequency = (highPassFilter != null) ? highPassFilter.cutoffFrequency : NeutralLowFrequency;
+
+            // Preallocate the array that will be used to collect RaycastHit structures.
+            hits = new RaycastHit[MaxObjects];
+        }
+
+        private void Update() 
+        {
+            DateTime now = DateTime.Now;
+
+            // Audio influences are not updated every frame.
+            if ((UpdateInterval * 1000.0f) <= (now - lastUpdate).Milliseconds)
+            {
+                audioSource.volume = initialAudioSourceVolume;
+
+                // Get the audio influencers that should apply to the audio source.
+                List<IAudioInfluencer> influencers = GetInfluencers();
+                foreach (IAudioInfluencer influencer in influencers)
+                {
+                    // Apply the influencer's effect.
+                    influencer.ApplyEffect(gameObject);
+                }
+
+                // Find and remove the audio influencers that are to be removed from the audio source.
+                List<IAudioInfluencer> influencersToRemove = new List<IAudioInfluencer>();
+                foreach (IAudioInfluencer prev in previousInfluencers)
+                {
+                    MonoBehaviour mbPrev = prev as MonoBehaviour;
+
+                    // Remove influencers that are no longer in line of sight
+                    // OR
+                    // Have been disabled
+                    if (!influencers.Contains(prev) ||
+                        ((mbPrev != null) && !mbPrev.isActiveAndEnabled))
+                    {
+                        influencersToRemove.Add(prev);
+                    }
+                }
+                RemoveInfluencers(influencersToRemove);
+
+                previousInfluencers = influencers;
+                lastUpdate = now;
+            }
+        }
+
+        private void OnValidate()
+        {
+            audioSource = gameObject.GetComponent<AudioSource>();
+        }
+
+        /// <summary>
+        /// Removes the effects applied by specified audio influencers.
+        /// </summary>
+        /// <param name="influencers">Collection of IAudioInfluencer objects on which to remove the effect.</param>
+        private void RemoveInfluencers(List<IAudioInfluencer> influencers)
+        {
+            foreach (IAudioInfluencer influencer in influencers)
+            {
+                influencer.RemoveEffect(gameObject);
+            }
+        }
+
+        /// <summary>
+        /// Finds the IAudioInfluencer objects that are to be applied to the audio source.
+        /// </summary>
+        /// <returns>Collection of IAudioInfluencers between the user and the game object.</returns>
+        private List<IAudioInfluencer> GetInfluencers()
```

Comment:
This could also be a property.

Reply:
technically, but it feels more like a method based on the work it is doing

Reply:
It is def a property `get` in the name says it all.

Reply:
should unity have implemented GetComponents as a property?  there are plenty of APIs that both have properties and Get* methods.

my bar is that properties should not do very much work (set a value, get a value, validate parameters). this is due to properties looking and feeling like a field... there should be minimal if any side effects

Reply:
this is how the .net framework was written and is a reasonably common engineering pattern

Reply:
> my bar is that properties should not do very much work (set a value, get a value, validate parameters). this is due to properties looking and feeling like a field... there should be minimal if any side effects

That seems to check all the boxes in this case. I'll leave it up to you.